### PR TITLE
Cleanup Soap Client

### PR DESCRIPTION
**Pull request by qw3ry**
Remove superfluous overload, add convenience methods for handling namespaces

### CODE REVIEW

File: src/main/java/sirius/kernel/xml/StructuredOutput.java [hunk 1]
```diff
@@ -71,6 +86,18 @@
...
 
+    /**
+     * Convenience method for {@link #property(String, Object)} prepending a namespace.
+     *
+     * @param namespace the namespace
+     * @param name      the name of the property
+     * @param data      the value of the property
+     * @return the output itself for fluent method calls
+     */
+    default StructuredOutput property(@Nonnull String namespace, @Nonnull String name, @Nullable Object data) {
```
Comment (andyHa) [inline src/main/java/sirius/kernel/xml/StructuredOutput.java:97]: shoudln't we move this into XMLStructuredOutput as JSON isn't namespace aware..

Comment (qw3ry) [reply src/main/java/sirius/kernel/xml/StructuredOutput.java:97]: Good point +1

File: src/main/java/sirius/kernel/xml/StructuredOutput.java [hunk 2]
```diff
@@ -38,14 +38,6 @@
      */
     void endResult();
 
-    /**
-     * Starts a new object with the given name.
-     *
-     * @param name the name of the element to start
-     * @return the output itself for fluent method calls
-     */
-    StructuredOutput beginObject(@Nonnull String name);
```
Comment (andyHa) [inline src/main/java/sirius/kernel/xml/StructuredOutput.java:47]: sure about this? this would be a breaking change - deprecate otherwise?

**Review by andyHa — COMMENTED** [review]
...

File: src/main/java/sirius/kernel/xml/StructuredOutput.java (hunk 2 above)
Comment (qw3ry) [reply src/main/java/sirius/kernel/xml/StructuredOutput.java:47]: It has an overload `StructuredOutput beginObject(@Nonnull String name, Attribute... attributes)` which will be called instead. A library user might need to recompile, but the code will still be compatible

**Review by andyHa — APPROVED** [review]

**Comment by qw3ry** [comment]
"Breaking" because two methods became one, and cannot both be overridden. I will adjust the corresponding code in -web once this PR is merged

**Review by bwiedmann — APPROVED** [review]

**Comment by qw3ry** [comment]
https://github.com/scireum/sirius-web/pull/747 required changes in web

**Review by meg1502 — APPROVED** [review]

**Review by andyHa — APPROVED** [review]

**Review by sabieber — APPROVED** [review]

**Review by mkeckmkeck — APPROVED** [review]

**Pull request merged**